### PR TITLE
Recipe-app:Create-Recipes-List-and-View

### LINKED ISSUE #7: [2pt] Create Recipes List View
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

### PR DESCRIPTION
**Pull request by Kumilachew-g**
**In this PR, I have done the following tasks.**

- Added recipe controller and implemented methods
- Added recipe_food controller and implement the methods
- Added recipe views
- Added recipe details view
- Added routes for the recipe and recipe_food controller implementation
- Added public recipe views

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,16 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
```
Comment (wandergithub) [inline app/controllers/application_controller.rb:4]: - [ ] Using `devise` gem here but not installed. Please add it to the Gemfile for proper configuration.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+class RecipesController < ApplicationController
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.order(:id).page(params[:page]).per(2)
+  end
```
Comment (wandergithub) [inline app/controllers/recipes_controller.rb:5]: The method `page` throws an error. And I can't make sense of what it should do. Can you explain it to me, so I can have more clarity?

Here is a screenshot of the error 
image

**Review by wandergithub — CHANGES_REQUESTED** [review]
### Please, make this changes

Hey @Kumilachew-g. Seems like you have been putting a lot of effort to the project.  That's awesome. But kindly, can you solve some of the problems that I point out in the comments, so I can test all of your code.

Also try merging the dev branch, so you are up-to-date with all the development code.

> Ask for a re-review when you are done.

File: app/controllers/recipes_controller.rb (hunk 2 above)
Comment (Kumilachew-g) [reply app/controllers/recipes_controller.rb:5]: I used this for pagination but I did not install "kaminari dem" library. For now, I will remove but We will implement it later. Thanks!!!

**Comment by Kumilachew-g** [comment]
**It looks like this...**
  
image

**Comment by Kumilachew-g** [comment]
Hi @wandergithub👋

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Kumilachew-g) [reply app/controllers/application_controller.rb:4]: I have fixed it. Thanks for your detailed review!!!

**Review by wandergithub — APPROVED** [review]
Now you can merge this branch 👍🏽

**Pull request merged**